File: frontend/src/types/macro.ts
export type MacroObservation = {
  date: string;
  value: number;
};

export type MacroSeries = {
  series_id: string;
  title: string;
  units: string;
  frequency: string;
  latest_value: number | null;
  previous_value: number | null;
  change: number | null;
  observations: MacroObservation[];
};

export type MacroContext = {
  available: boolean;
  reason: string | null;
  market_id: number;
  market_title: string;
  series: MacroSeries[];
  macro_source: string | null;
};

File: backend/app/schemas/macro.py
from datetime import date

from pydantic import BaseModel


class MacroObservationRead(BaseModel):
    date: date
    value: float


class MacroSeriesRead(BaseModel):
    series_id: str
    title: str
    units: str
    frequency: str
    latest_value: float | None = None
    previous_value: float | None = None
    change: float | None = None
    observations: list[MacroObservationRead]


class MacroContextResponse(BaseModel):
    available: bool
    reason: str | None = None
    market_id: int
    market_title: str
    series: list[MacroSeriesRead]
    macro_source: str | None = None
